### PR TITLE
Fix user created another user account when logged in through google after having a verified email

### PR DESCRIPTION
**Pull request by valentinludu**
### <samp>🤖 Generated by Copilot at 863aa15</samp>

This pull request adds support for logging in with Google when the user has an existing verified email in the database. It modifies the `loginWithGoogle` function in `lib/google/loginWithGoogle.ts` and adds a test case in `lib/google/__tests__/loginWithGoogle.spec.ts`.

### WHY
An user lost access to their account by doing:
Create user and space with just their email (verifiedEmail) -> Logout -> Login with Google. 
The app created another user with no space.

[Task item](https://app.charmverse.io/charmverse/page-9248424773658832)

### CODE REVIEW

File: lib/google/loginWithGoogle.ts [hunk 1]
```diff
@@ -69,6 +82,30 @@ export async function loginWithGoogle({
       }
     }
 
+    if (emailAccount) {
+      if (emailAccount.user.deletedAt) {
+        throw new DisabledAccountError(`This account has been disabled`);
```
Comment (mattcasey) [inline lib/google/loginWithGoogle.ts:87]: i dont think this should be an error....? could we just ignore the user

**Review by mattcasey — CHANGES_REQUESTED** [review]
i don't think a previously deleted user should start failing this case.. maybe just log a warning?

File: lib/google/loginWithGoogle.ts (hunk 1 above)
Comment (valentinludu) [reply lib/google/loginWithGoogle.ts:87]: OK. I can ignore that. 

I was just taking the example from above (google account)

Comment (valentinludu) [reply lib/google/loginWithGoogle.ts]: Pushed update!

**Review by mattcasey — APPROVED** [review]

**Pull request merged**